src/components/ha-sidebar.ts

```diff
@@ -285,6 +294,10 @@ class HaSidebar extends SubscribeMixin(LitElement) {
   protected firstUpdated(changedProps: PropertyValues) {
     super.firstUpdated(changedProps);
     this._subscribePersistentNotifications();
+
+    window.addEventListener("hass-reset-sidebar", (ev) => {
+      this._sidebar.selected = ev.detail;
+    });
```

Comment:
This does not work in my tests

Reply:
I think the better and easier solution here is adding:

```ts
        @iron-activate=${preventDefault}
```

to the `paper-listbox` element. This will prevent the listbox setting the selected element on click, and will force it to only update when `hass.panelUrl` has updated. Which is exactly what we want.